### ISSUE TITLE
Add support for the non-standard REPLACE statement

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#547]
Some RDBMS implement a `REPLACE` clause that is in fact an `INSERT` preceded by an optional `DELETE`. See an example:

http://dev.mysql.com/doc/refman/5.5/en/replace.html

**Comment by jklingsporn** [comment]
I'm currently substituting this with a INSERT ... ON DUPLICATE KEY UPDATE statement. However the value I update is a big blob. Using the REPLACE statement would save some bandwidth here.

**Comment by lukaseder** [comment]
That's true. If bandwidth is a significant issue for a few queries, you could also write your own `REPLACE` statement support for the time being. Something along the lines of:

```java
public static int executeReplace(Table<?> table, Field<?>[] columns, Object[] values) {
    return DSL.execute("replace into {0} ({1}) values ({2})", 
        table, 
        DSL.list(columns), 
        DSL.list(Stream.of(values).map(DSL::val).collect(toList()))
    );
}
```

**Comment by lukaseder** [comment]
> I'm currently substituting this with a INSERT ... ON DUPLICATE KEY UPDATE statement. However the value I update is a big blob. Using the REPLACE statement would save some bandwidth here.

The bandwidth problem is avoided by using the PG-style `excluded()` function: https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/impl/DSL.html#excluded(org.jooq.Field)

Which translates to `VALUES(x)` or `NEW.x` in MySQL:
https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html

Other than that, I don't really see much value in this `REPLACE` statement. It's just a more convenient `MERGE` (similar to H2's old `MERGE` statement, which we deprecated: 
https://www.h2database.com/html/commands.html#merge_into

Specifically, deleting and inserting does seem more costly in the DB than just updating, at least for some workloads.
